uxarray/grid/centroids.py

```diff
@@ -0,0 +1,64 @@
+import xarray as xr
+import numpy as np
+from uxarray.grid.coordinates import node_xyz_to_lonlat_rad, node_lonlat_rad_to_xyz
+
+
+def centroid_from_mean_verts(self, repopulate=True):
+    """Finds the centroids of faces based on the mean value of the vertices."""
+
+    latlon = False
+    if "Mesh2_node_z" not in self._ds:
+        latlon = True
+        lon = self.Mesh2_node_x.values
+        lat = self.Mesh2_node_y.values
+        [node_x, node_y, node_z] = node_lonlat_rad_to_xyz([lon, lat])
+    else:
+        node_x = self.Mesh2_node_cart_x.values
+        node_y = self.Mesh2_node_cart_y.values
+        node_z = self.Mesh2_node_cart_z.values
```

Comment:
We shouldn't need these checks, since directly calling `self.Mesh2_node_cart_x/y/z` should check and construct them on call. 

Example Below

https://github.com/UXARRAY/uxarray/blob/ba113ae96de1d2931a2d83336b8401f7e008ab68/uxarray/grid/grid.py#L356-L366